### ISSUE TITLE
Knife-edge coronagraph on different sides

### ISSUE THREAD

**Issue by syhaffert** [ehpor/hcipy#111]
I just pushed a branch where I added different directions of the Knife-edge coronagraph. It works and I have added tests. However, there is a slight asymmetry in the peak flux after the knife-edge between the 'left' and 'right' versions. It is the same for the 'up' and 'down' direction. 

Here is a minimal working example that demonstrates this. I am not sure why this happens. I checked both even and odd grids, so it does not seem to be a symmetry issue due to the inclusion of a zero in the Fourier transform. The shape coronagraphic residuals do look similar.

```
from hcipy import *
import numpy as np
from matplotlib import pyplot as plt

grid = make_pupil_grid([256, 512])
aperture = circular_aperture(1)(grid)

focal_grid = make_focal_grid(q=5, num_airy=15)
prop = FraunhoferPropagator(grid, focal_grid)

lyot_stop = circular_aperture(0.95)(grid)

wf = Wavefront(aperture)
norm = prop(wf).power.max()

directions = ['left', 'right', 'down', 'up']
for i, direction in enumerate(directions):
if direction == 'left':
	pre_apodizer = np.exp(1j * 2 * np.pi * grid.x * -1)
elif direction == 'right':
	pre_apodizer = np.exp(1j * 2 * np.pi * grid.x * 1)
elif direction == 'up':
	pre_apodizer = np.exp(1j * 2 * np.pi * grid.y * 1)
else:
	pre_apodizer = np.exp(1j * 2 * np.pi * grid.y * -1)

knife = KnifeEdgeLyotCoronagraph(grid, knife_edge_side=direction, apodizer=pre_apodizer, lyot_stop=lyot_stop * np.conj(pre_apodizer))
wf_foc = prop(knife(wf))
print(wf_foc.power.max() / norm)

plt.subplot(2,2,i+1)
imshow_psf(wf_foc.power / norm, vmax=1e-2, vmin=1e-5)
plt.show()
```

**Comment by syhaffert** [comment]
I think I found the issue. The knife-edge cor. implementation uses 1D FFTs with over sampling q. The peak intensities become the same if I apply an additional correction of 1/q to the pre-apodizer. This is a single pixel shift issue. I have added the new code that creates the correct result. 

`from hcipy import *
import numpy as np
from matplotlib import pyplot as plt


grid = make_pupil_grid([257, 513])
aperture = circular_aperture(1)(grid)

focal_grid = make_focal_grid(q=5, num_airy=15)
prop = FraunhoferPropagator(grid, focal_grid)

lyot_stop = circular_aperture(0.95)(grid)

wf = Wavefront(aperture)
norm = prop(wf).power.max()

directions = ['left', 'right', 'down', 'up']
psfs = []
indx = [0, 1, 3, 4]
qtest = 16
for i, direction in enumerate(directions):
	if direction == 'left':
		pre_apodizer = np.exp(1j * 2 * np.pi * grid.x * (-1 - 1/qtest))
	elif direction == 'right':
		pre_apodizer = np.exp(1j * 2 * np.pi * grid.x * 1)
	elif direction == 'up':
		pre_apodizer = np.exp(1j * 2 * np.pi * grid.y * 1)
	else:
		pre_apodizer = np.exp(1j * 2 * np.pi * grid.y * (-1 - 1/qtest))

	knife = KnifeEdgeLyotCoronagraph(grid, q=qtest, knife_edge_side=direction, apodizer=pre_apodizer, lyot_stop=lyot_stop * np.conj(pre_apodizer))
	wf_foc = prop(knife(wf))
	print(wf_foc.power.max() / norm)
	psfs.append(wf_foc.power / norm)

	plt.subplot(2, 3, indx[i] + 1)
	imshow_psf(wf_foc.power / norm, vmax=1e-2, vmin=1e-5)

dI = psfs[0] - psfs[1].shaped[:, ::-1].ravel()
plt.subplot(2, 3, 3)
imshow_field(dI, vmax=1e-2, vmin=1e-5)

dI = psfs[2] - psfs[3].shaped[::-1,:].ravel()
plt.subplot(2, 3, 6)
imshow_field(dI, vmax=1e-2, vmin=1e-5)

plt.show()`

**Comment by ehpor** [comment]
So then the knife edge itself is not centered... Hmm... I thought https://github.com/ehpor/hcipy/blob/d5e0438ef3fd5756f27cef7d6f704eb961c98496/hcipy/coronagraphy/knife_edge.py#L46 took care of that.

**Comment by syhaffert** [comment]
Shouldn't it be sqrt(0.5)? Because the electric field transmission should be the sqrt of the transmission?

**Comment by syhaffert** [comment]
solved by PR #142 . I will close this issue.
